apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/V1/SessionPoolTests.TargetedSessionPoolTests.cs

```diff
@@ -299,6 +299,226 @@ public async Task AcquireAsync_MaxActiveSessions_Block()
+                // The extra ones should be pending waiting on create session tasks.
+                var acquisitionTasks = Enumerable
+                    .Range(0, 13)
+                    .Select(_ => pool.AcquireSessionAsync(new TransactionOptions(), default))
+                    .ToList();
+                // Wait a little in real time to make sure that session creation tasks have started.
+                // Session creation is done in a (controlled) fire and forget way, that's why
+                // we need to wait a little in real time.
+                await Task.Delay(250);
+                stats = pool.GetStatisticsSnapshot();
+                Assert.True(stats.Healthy);
+                Assert.Equal(13, stats.ActiveSessionCount);
+                Assert.Equal(3, stats.PendingAcquisitionCount);
+                Assert.Equal(0, stats.ReadPoolCount);
+                Assert.Equal(0, stats.ReadWritePoolCount);
+                Assert.Equal(13, stats.InFlightCreationCount);
+
+                // Make the create session tasks fail and notify the waiting callers.
+                await client.Scheduler.RunAsync(TimeSpan.FromSeconds(5));
+
+                int succesfulAcquisitions = 0;
+                int failedAcquisitions = 0;
+                while(acquisitionTasks.Count > 0)
+                {
+                    var acquisitionTask = await Task.WhenAny(acquisitionTasks);
+                    acquisitionTasks.Remove(acquisitionTask);
+
+                    // It can either be one that got a previously created session
+                    // or one that failed.
+                    try
+                    {
+                        await acquisitionTask;
+                        succesfulAcquisitions++;
+                    }
+                    catch
+                    {
+                        failedAcquisitions++;
+                    }
+                }
+                Assert.Equal(10, succesfulAcquisitions);
+                Assert.Equal(3, failedAcquisitions);
+                stats = pool.GetStatisticsSnapshot();
+                Assert.False(stats.Healthy);
+                Assert.Equal(10, stats.ActiveSessionCount);
+                Assert.Equal(0, stats.PendingAcquisitionCount);
+                Assert.Equal(0, stats.ReadPoolCount);
+                Assert.Equal(0, stats.ReadWritePoolCount);
+                Assert.Equal(0, stats.InFlightCreationCount);
+            }
+
+            [Fact(Timeout = TestTimeoutMilliseconds)]
+            public async Task AcquireAsync_NursesBackToHealth()
+            {
+                var pool = CreatePool(false);
+                var client = (SessionTestingSpannerClient)pool.Client;
+                var options = pool.Options;
+
+                client.FailAllRpcs = true;
+                client.FailAllRpcsDelay = TimeSpan.FromSeconds(1);
+
+                // In total MinimumSizePool + 1 sessions will be created (or tried to in our case)
```

Comment:
Is the +1 because of the "trying to nurse back to health"? If so, that's probably worth saying in the comment.